### PR TITLE
UploadRequest to provide original xhr request as part of callback par…

### PR DESCRIPTION
**Pull request by manish-tal**
…ameters

### CODE REVIEW

**Comment by zackify** [comment]
Why is this needed?

**Comment by manish-tal** [comment]
Because I am sending some information in response header and I need that information after files are done uploading.

File: src/request/register-listeners.js [hunk 1]
```diff
@@ -11,10 +11,21 @@ export default ({ xhr, progress, resolve }) => {
     } catch (e) {
       response = xhr.response;
     }
+    var headers = xhr.getAllResponseHeaders();
```
Comment (audiolion) [inline src/request/register-listeners.js]: A bit more concise way:
```js
let headers = xhr
  .getAllResponseHeaders()
  .trim()
  .split(/[\r\n]+/)
  .map(line => line.split(': '))
  .reduce((acc, [header, value]) => {
    acc[header] = value;
    return acc;
  }, {});
```

Comment (manish-tal) [reply src/request/register-listeners.js]: that is even better

**Comment by manish-tal** [comment]
I have added test cases for headers please review

**Comment by zackify** [comment]
Thanks! unfortunately it looks like i dont have permission on this repo. hopefully they grant it back to me, then i can merge

**Comment by zackify** [comment]
This is now up! Thanks https://github.com/ConstructionJobs/upload/releases/tag/3.2.0

**Pull request merged**